src/UnixConnector.php

```diff
@@ -27,7 +28,12 @@ public function connect($path)
         if (strpos($path, '://') === false) {
             $path = 'unix://' . $path;
         } elseif (substr($path, 0, 7) !== 'unix://') {
-            return Promise\reject(new \InvalidArgumentException('Given URI "' . $path . '" is invalid'));
+            return Promise\reject(new InvalidArgumentException('Given URI "' . $path . '" is invalid'));
+        }
+
+        $maxLen = PHP_OS === 'Darwin' ? 104 : 108;
+        if (strlen($path) > $maxLen) {
+            return Promise\reject(new InvalidArgumentException('Given URI "' . $path . '" exceeds maximum allowed length'));
```

Comment:
I think you're addressing a relevant issue here, thank you! However, I would both get a better understanding of this problem before getting this in and also believe that this is unrelated to this new feature. Does it make sense to you to remove this for now and file this a separate PR? :+1: 

Some thoughts (short brain dump only):
* The path limit seems to apply to the path as given, a prior `chdir()` may be used to work around this for use use cases
* The `unix://` scheme is not included in this limit
* The limit is system dependent and 108 for Linux and 104 for BSD (Mac) and lower for other systems
* Linux man pages suggest `108` is a valid length, but PHP already raises a notice here
* The E_NOTICE was apparently introduced via https://bugs.php.net/bug.php?id=60106 in PHP 5.4.1+
* All relevant versions (including < PHP 5.4.1) already (silently) truncated the path
* My experiments: https://3v4l.org/hSRCU

Reply:
Thanks for the experiments :)

> The path limit seems to apply to the path as given, a prior chdir() may be used to work around this for use use cases

Would need to double-check.

> The unix:// scheme is not included in this limit

Why do you think so? The previous unit tests seemed to indicate that it does?

> The limit is system dependent and 108 for Linux and 104 for BSD (Mac) and lower for other systems

Unless we could identify from some setting which those are we could only live with assumptions. Custom kernels might have yet other limits?

> Linux man pages suggest 108 is a valid length, but PHP already raises a notice here

I can only imagine PHP takes the trailing 0 into account for that check. Might be a bug?

> All relevant versions (including < PHP 5.4.1) already (silently) truncated the path

Meaning your suggestion would be to accept this situation? 

I'd be happy to not re-raise this PR but thought it would be good as a precaution as I've only accidentally hit this limit and it might have gone unnoticed.

Long story short: do you think it is worth pursuing this or rather entirely skip it?